<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1321</width>
    <height>798</height>
   </rect>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Path Planning for Modelling Natural Phenomena 1.0</string>
  </property>
  <widget class="QWidget" name="m_centralwidget">
   <layout class="QGridLayout" name="s_mainGridLayout">
    <item row="0" column="2" rowspan="7">
     <layout class="QGridLayout" name="gridLayout">
      <item row="5" column="0" rowspan="2">
       <widget class="QGroupBox" name="groupBox_5">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>176</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Step 3 - Select End Points</string>
        </property>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>30</y>
           <width>351</width>
           <height>151</height>
          </rect>
         </property>
         <property name="title">
          <string>Random Select</string>
         </property>
         <widget class="QWidget" name="gridLayoutWidget_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>20</y>
            <width>351</width>
            <height>128</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Max</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QDoubleSpinBox" name="m_yMin">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="value">
              <double>-5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QDoubleSpinBox" name="m_zMax">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="value">
              <double>5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QDoubleSpinBox" name="m_zMin">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="value">
              <double>-5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="m_xMax">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="value">
              <double>5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Amount</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="m_xMin">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="value">
              <double>-5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QDoubleSpinBox" name="m_yMax">
             <property name="minimum">
              <double>-99.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="value">
              <double>5.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Min</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="5">
            <widget class="QPushButton" name="pushButton_2">
             <property name="text">
              <string>Select Random Endpoints</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1" colspan="4">
            <widget class="QSpinBox" name="m_randomEndPointCount">
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item row="1" column="4" rowspan="2">
            <widget class="QPushButton" name="m_previewBoundingBox">
             <property name="text">
              <string>P</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
        <widget class="QGroupBox" name="groupBox_7">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>180</y>
           <width>351</width>
           <height>201</height>
          </rect>
         </property>
         <property name="title">
          <string>Index Select</string>
         </property>
         <widget class="QWidget" name="gridLayoutWidget_4">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>20</y>
            <width>351</width>
            <height>29</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="1">
            <widget class="QLineEdit" name="m_endpointIndexLineEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>24</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>24</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Endpoint</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="m_addEndPointPushButton">
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>24</height>
              </size>
             </property>
             <property name="text">
              <string>+</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="gridLayoutWidget_5">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>50</y>
            <width>351</width>
            <height>135</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="1">
            <widget class="QListWidget" name="m_endpointIndicesList">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>100</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="m_deleteEndPointPushButton">
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>24</height>
              </size>
             </property>
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QPushButton" name="m_generatePathsPushButton">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>26</height>
              </size>
             </property>
             <property name="text">
              <string>Generate Paths</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>80</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>80</height>
         </size>
        </property>
        <property name="title">
         <string>Step 2 - Select Root</string>
        </property>
        <widget class="QWidget" name="gridLayoutWidget_6">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>351</width>
           <height>62</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="0">
           <widget class="QLabel" name="m_rootIndexLabel">
            <property name="text">
             <string>Root Index</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="m_rootIndexSpinBox">
            <property name="maximum">
             <number>10000</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QPushButton" name="m_calcDistancesPushButton">
            <property name="text">
             <string>Calculate Distances From Root</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox">
        <property name="minimumSize">
         <size>
          <width>360</width>
          <height>235</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>240</height>
         </size>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="title">
         <string>Step 1 - Construct Weighted Lattice</string>
        </property>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>30</y>
           <width>481</width>
           <height>81</height>
          </rect>
         </property>
         <property name="title">
          <string>Lattice Options</string>
         </property>
         <widget class="QWidget" name="gridLayoutWidget">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>20</y>
            <width>351</width>
            <height>62</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="3">
            <widget class="QSpinBox" name="m_nSpinBox">
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="displayIntegerBase">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Irregular</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="m_distanceLabel">
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Edge Length</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="m_latticeComboBox">
             <property name="frame">
              <bool>true</bool>
             </property>
             <item>
              <property name="text">
               <string>Square 2D</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Hexagon 2D</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Cube</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="m_typeLabel">
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Type</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QCheckBox" name="m_isIrregular">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="m_distanceDoubleSpinBox">
             <property name="maximum">
              <double>5.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="m_nLabel">
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Steps</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>120</y>
           <width>351</width>
           <height>121</height>
          </rect>
         </property>
         <property name="title">
          <string>Weight Function Options</string>
         </property>
         <widget class="QWidget" name="gridLayoutWidget_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>20</y>
            <width>351</width>
            <height>97</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="1">
            <widget class="QComboBox" name="m_weightFunctionTypeComboBox">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Default (eR^alpha)</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Image (2D Lattices Only)</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="m_filenameLabel">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Filename</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="4">
            <widget class="QPushButton" name="m_constructLatticePushButton">
             <property name="text">
              <string>Construct Lattice</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" colspan="3">
            <layout class="QHBoxLayout" name="m_filebrowseHorizontalLayout">
             <item>
              <widget class="QLineEdit" name="m_filenameLineEdit">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="m_browsePushButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximumSize">
                <size>
                 <width>30</width>
                 <height>27</height>
                </size>
               </property>
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="m_weightFunctionLabel">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Type</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="m_alphaLabel">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Alpha</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QDoubleSpinBox" name="m_alphaDoubleSpinBox">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="maximum">
              <double>5.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item row="6" column="0">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="6" column="1">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1321</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExportPaths"/>
    <addaction name="separator"/>
    <addaction name="actionExitApp"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionRefinePaths"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionW"/>
   <addaction name="actionI"/>
   <addaction name="actionN"/>
   <addaction name="actionD"/>
   <addaction name="actionL"/>
   <addaction name="actionB"/>
  </widget>
  <action name="actionExportPaths">
   <property name="text">
    <string>Export Paths</string>
   </property>
  </action>
  <action name="actionExitApp">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionW">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>W</string>
   </property>
   <property name="toolTip">
    <string>Toggle Weights</string>
   </property>
  </action>
  <action name="actionI">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>I</string>
   </property>
   <property name="toolTip">
    <string>Toggle Reference Image</string>
   </property>
  </action>
  <action name="actionN">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>N</string>
   </property>
   <property name="toolTip">
    <string>Toggle Node Indices</string>
   </property>
  </action>
  <action name="actionD">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>D</string>
   </property>
   <property name="toolTip">
    <string>Toggle Distances</string>
   </property>
  </action>
  <action name="actionL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>L</string>
   </property>
   <property name="toolTip">
    <string>Toggle Lattice</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
  </action>
  <action name="actionRefinePaths">
   <property name="text">
    <string>Refine Paths</string>
   </property>
  </action>
  <action name="actionB">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>B</string>
   </property>
   <property name="toolTip">
    <string>Toggle Endpoint Bounding Box</string>
   </property>
   <property name="shortcut">
    <string>B</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
